lib/perfdata/elasticsearchwriter.cpp

```diff
@@ -131,6 +132,40 @@ void ElasticsearchWriter::Pause()
 	ObjectImpl<ElasticsearchWriter>::Pause();
 }
 
+void ElasticsearchWriter::AddTemplateTags(const Dictionary::Ptr& fields, const Checkable::Ptr& checkable, const CheckResult::Ptr& cr)
+{
+	Host::Ptr host;
+	Service::Ptr service;
+	tie(host, service) = GetHostService(checkable);
+
+	Dictionary::Ptr tmpl;
+	if (service) {
+		if (GetServiceTagsTemplate()) {
+			tmpl = static_pointer_cast<Dictionary>(GetServiceTagsTemplate()->ShallowClone());
+		}
+	} else if (GetHostTagsTemplate()) {
+		tmpl = static_pointer_cast<Dictionary>(GetHostTagsTemplate()->ShallowClone());
+	}
```

Comment:
After working on different stuff for some time I also had to reread into it and you are absolutely right here.

As stated earlier I worked with the influxdb writer as a model.
There the returned Template is also used to store the resolved values and so cloning makes sense.
But since we only ever read from the retrieved template we do not have to clone it.